articles/spring-data-mongodb/index.md

```diff
@@ -0,0 +1,170 @@
+- **Embedded document** is an efficient and clean way to store related data, especially data that are regularly accessed together
+- **Primary key** uniquely identifies a document.
+
+### Creating application
+We will be using [spring initializr](https://start.spring.io/) to create our application.
+1. Open [Spring initializr](https://start.spring.io/) in your browser.
+2. Select language to Kotlin.
+3. Add `Spring Web`, `Spring Data MongoDB`, dependencies.
+4. Leave other configurations as default and click on generate the project.
+5. Unzip the downloaded project and open it in your favorite IDE. I use [Intelij IDEA community](https://www.jetbrains.com/idea/download/#section=linux) that is available for free.
+6. Sync the project with maven to download all the dependencies.
+
+
+### Domain layer
+In the root package of our application, create a package named `domain`.
+Create a kotlin file named `Todo.kt` in the domain package we have created above and add the code snippet below into the file.
+```kotlin
+import org.springframework.data.annotation.Id
+import org.springframework.data.mongodb.core.mapping.Document
+
+@Document(collection = "todo")
+data class Todo(
+    @Id
+    val id: String,
+    var title: String,
+    var description: String,
+    var isCompleted: Boolean
+)
+```
+- `@Document(collection = "todo")` annotation marks the `Todo` data class as a MongoDB document that will be stored under the `todo` collection.
+- `@Id` marks the `id` field as a unique object id in the MongoDB collection.
+
+### Controller layer
+In the root package of our application, create a package named `controllers`.
+Create a kotlin file named `TodoController.kt` and add the code snippet below.
+```kotlin
+@RestController
+@RequestMapping("/api/v1/todos/")
+@CrossOrigin("*")
+class TodoController(val todoService: TodoService) {
+    //Get all todos in the database
+    @GetMapping("")
+    fun getAllTodos(): ResponseEntity<List<Todo>> {
+        return ResponseEntity(todoService.getTodos(), HttpStatus.OK)
+    }
+
+    //Get Todo from the datababse by id
+    @GetMapping("/{id}")
+    fun getTodoById(@PathVariable("id") id: String): ResponseEntity<Todo> {
+        return ResponseEntity(todoService.getTodoById(id), HttpStatus.OK)
+    }
+
+    //save a todo into the database
+    @PostMapping("")
+    fun saveTodo(@RequestBody todo: Todo): ResponseEntity<Any> {
+        return ResponseEntity(todoService.saveTodo(todo), HttpStatus.CREATED)
+    }
+
+    //update todo
+    @PutMapping("")
+    fun updateTodo(@RequestBody todo: Todo): ResponseEntity<Any> {
+        return ResponseEntity(todoService.updateTodo(todo), HttpStatus.OK)
+    }
+
+    //delete todo
+    @DeleteMapping("/{id}")
+    fun deleteTodo(@PathVariable("id") id: String): ResponseEntity<Any> {
+        return ResponseEntity(todoService.deleteTodo(id), HttpStatus.NO_CONTENT)
+    }
+
+}
+```
+- `@CrossOrigin` annotation is used to configure the allowed origins.
```

Comment:
You should also mention that it is the same as cors for the developers not experienced with cross origin issues